packages/codemirror-extension/src/services.tsx

```diff
@@ -148,16 +148,30 @@ export const extensionPlugin: JupyterFrontEndPlugin<IEditorExtensionRegistry> =
 
         formRegistry?.addRenderer(`${SETTINGS_ID}.defaultConfig`, {
           fieldRenderer: (props: FieldProps) => {
+            let defaultFormData: Record<string, any>;
             const properties = React.useMemo(
               () => registry.settingsSchema,
               []
             ) as any;
-            const defaultFormData: Record<string, any> = {};
+            if (
+              [
+                'codeCellConfig',
+                'markdownCellConfig',
+                'rawCellConfig'
+              ].includes(props.name)
+            ) {
+              defaultFormData = props.formContext.defaultFormData[props.name];
```

Comment:
I wonder if we can find a way to implement this without hard-coding any specific IDs...